normandy/classifier/tests/test_api.py

```diff
@@ -52,3 +55,22 @@ def test_it_filters_by_locale(self, client):
 
         recipe_names = set(r['name'] for r in res.data['recipes'])
         assert recipe_names == {'german', 'any', 'both'}
+
+    @pytest.mark.xfail
+    def test_it_makes_no_db_queries(self, client, django_cache):
+        # Prepare some interesting data
+        recipe_action = RecipeActionFactory(recipe__enabled=True)
+        recipe = recipe_action.recipe
+
+        # Warm up the cache
+        res1 = client.post('/api/v1/fetch_bundle/', {'locale': 'de'})
+        assert res1.status_code == 200
+        assert res1.data['recipes'][0]['name'] == recipe.name
+
+        # Fire!
+        queries = CaptureQueriesContext(connection)
```

Comment:
It is! It is also a little verbose. I think I might want to add a little helper that makes it more straightforward. For example, not needing to grab the Django DB connection object would be nice.